qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -0,0 +1,245 @@
+    EstimatorGradientResult,
+)
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator, Estimator, EstimatorResult
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError
+
+from .neural_network import NeuralNetwork
+
+logger = logging.getLogger(__name__)
+
+
+class EstimatorQNN(NeuralNetwork):
+    """A Neural Network implementation based on the Estimator primitive.
+
+    The ``EstimatorQNN`` is a neural network that takes in a parametrized quantum circuit
+    with the combined network's feature map (input parameters) and ansatz (weight parameters)
+    and outputs its measurements for the forward and backward passes.
+
+    Attributes:
+
+        estimator (BaseEstimator): The estimator primitive used to compute the neural network's results.
+        gradient (BaseEstimatorGradient): An optional estimator gradient to be used for the backward
+            pass.
+
+    A Neural Network implementation based on the Estimator primitive."""
+
+    def __init__(
+        self,
+        *,
+        estimator: BaseEstimator | None = None,
+        circuit: QuantumCircuit,
+        observables: Sequence[BaseOperator | PauliSumOp] | BaseOperator | PauliSumOp | None = None,
+        input_params: Sequence[Parameter] | None = None,
+        weight_params: Sequence[Parameter] | None = None,
+        gradient: BaseEstimatorGradient | None = None,
+        input_gradients: bool = False,
+    ):
+        r"""
+        Args:
+            estimator: The estimator used to compute neural network's results.
+                If ``None``, a default instance of the reference estimator,
+                :class:`~qiskit.algorithms.Estimator`, will be used.
+            circuit: The quantum circuit to represent the neural network.
+            observables: The observables for outputs of the neural network. If ``None``,
+                use the default :math:`Z^{\otimes num\_qubits}` observable.
+            input_params: The parameters that correspond to the input data of the network.
+                If ``None``, the input data is not bound to any parameters.
+            weight_params: The parameters that correspond to the trainable weights.
+                If ``None``, the weights are not bound to any parameters.
+            gradient: The estimator gradient to be used for the backward pass.
+                If None, a default instance of the estimator gradient,
+                :class:`~qiskit.algorithms.ParamShiftEstimatorGradient`, will be used.
+            input_gradients: Determines whether to compute gradients with respect to input data.
+                Note that this parameter is ``False`` by default, and must be explicitly set to
+                ``True`` for a proper gradient computation when using ``TorchConnector``.
```

Comment:
Do we want to change TorhcConnector here to a class ref. so it becomes a link rather than just a highlighted term